Add method modeling flow to test plan

norascheuch opened the pull request:
<!-- Thank you for submitting a pull request. Please read our pull request guidelines before
  submitting your pull request:
  https://github.com/github/vscode-codeql/blob/main/CONTRIBUTING.md#submitting-a-pull-request.
-->

This PR adds checks for manually testing the new method modeling flow to our release test plan.


## Checklist

- [ ] [CHANGELOG.md](https://github.com/github/vscode-codeql/blob/main/extensions/ql-vscode/CHANGELOG.md) has been updated to incorporate all user visible changes made by this pull request.
- [ ] Issues have been created for any UI or other user-facing changes made by this pull request.
- [ ] _[Maintainers only]_ If this pull request makes user-facing changes that require documentation changes, open a corresponding docs pull request in the [github/codeql](https://github.com/github/codeql/tree/main/docs/codeql/codeql-for-visual-studio-code) repo and add the `ready-for-doc-review` label there.

norascheuch commented:
I wonder how many of these checks should be mandatory, what can be optional.
At the moment we're not testing the method modeling panels multiple method behaviour. Should this be mandatory/optional?

koesie10 commented on docs/test-plan.md:
```diff
@@ -152,6 +152,22 @@ Run one of the above MRVAs, but cancel it from within VS Code:
+      - add a couple of new models for one method
+      - save and check that the modeling file (use the 'open extension pack' button to open it) shows multiple methods
+      - check that a 'duplicated classification' error appears in both model editor and modeling panel when a duplicate modeling occurs
+      - check that a 'conflicting classigication' error appears when a neutral model type is paired with a source kind
```
Neutral models aren't conflicting with source kinds only, they are conflicting with models that have the same `type` as the neutral model's `kind`. What about:

```suggestion
      - check that a 'conflicting classification' error appears when a neutral model type is paired with a model of the same kind
```

koesie10 commented on docs/test-plan.md:
```diff
@@ -152,6 +152,22 @@ Run one of the above MRVAs, but cancel it from within VS Code:
+      - check that clicking on the error highlights the correct modeling in both the editor and the modeling panel
+3. Check the Method Usage
+   - Check that the Method Usage Panel opens and jumps to the correct usage when clicking on 'View' in the model editor
+   - Check that the usages are opening when clicking on a usage
```
We had a bug here where the first one would show correctly, but clicking on any other ones would still only show the first one. Would it make sense to change this to something like:

```suggestion
   - Check that the second and third usages show when clicking on them
```

koesie10 reviewed (commented):
Thanks for adding this! Multiple models are still feature-flagged right now, but since it is going to be removed soon, it's probably fine to not mention that here.

I'm not sure whether these should be optional or not. It's probably good to test them when we have made any changes to them, but don't need to test them when we've only made changes to unrelated code.

robertbrignull reviewed (commented):
I agree with Koen's points. But other than that looks good to me

norascheuch commented:
I implemented Koens suggestions and also moved most test cases to optional. In the mandatory testing we now only check that both Panels exist, show the first usage (usage panel) and show and update modeling (modeling panel).

robertbrignull commented on docs/test-plan.md:
```diff
@@ -189,9 +192,28 @@ Are there any components that are not showing up?
-### Selecting repositories to run on
+### Modeling Flow
+
+2. Check that a method can have multiple models:
```
The linter has spotted this needs to start with `1`, though I believe it won't actually affect the rendering it'll be nice to be consistent.

robertbrignull reviewed (approved):
There's a tiny linter error, but otherwise the changes LGTM

Thanks for moving stuff into "optional". I think that's the right move for these extra tests of the model editor.
